pkg/readiness/object_tracker.go

```diff
@@ -257,19 +257,24 @@ func (t *objectTracker) Populated() bool {
 // Expectations must be populated before the tracker can be considered satisfied.
 // Expectations are marked as populated by calling ExpectationsDone().
 func (t *objectTracker) Satisfied() bool {
+	// Determine if we need access to the write lock, a precious resource
```

Comment:
"precious" might be going a bit far. The difference is that write locks are globally exclusive. Only one goroutine can access a write-locked resource (both reads and writes blocked). Multiple goroutines can access a read lock at the same time (only writes are blocked).

"determine if we need to acquire a write lock, which blocks concurrent access" is more on-point (and less scary sounding)

Reply:
Done